src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +27,115 @@ interface GetCourseDetailsResponse {
-  }
-
-  get allCheckedStudentDetails(): StudentDetails[] {
-    return this.allPresentStudentDetails.filter((studentDetails: StudentDetails) => studentDetails.isChecked);
-  }
-
-  get isAllPresentCoursesChecked(): boolean {
-    return !!this.allPresentCourses.length && this.allPresentCourses.length === this.allCheckedCourses.length;
-  }
+  courseList!: Courses;
+  viewStudent!: boolean[];
+  courseInfo: {[key: string]: StudentListSectionData[]} = {};
+  courseStats: {[key: string]: Statistic} = {};
 
-  get isAllPresentCourseDetailsChecked(): boolean {
-    return !!this.allPresentCourseDetails.length
-        && this.allPresentCourseDetails.length === this.allCheckedCourseDetails.length;
+  constructor(private httpRequestService: HttpRequestService,
+              private courseService: CourseService,
+              private studentService: StudentService,
+              private statusMessageService: StatusMessageService,
+              private route: ActivatedRoute) {
   }
 
-  get isAllPresentSectionDetailsChecked(): boolean {
-    return !!this.allPresentSectionDetails.length
-        && this.allPresentSectionDetails.length === this.allCheckedSectionDetails.length;
-  }
-
-  get isAllPresentTeamDetailsChecked(): boolean {
-    return !!this.allPresentTeamDetails.length
-        && this.allPresentTeamDetails.length === this.allCheckedTeamDetails.length;
-  }
-
-  get isAllPresentStudentDetailsChecked(): boolean {
-    return !!this.allPresentStudentDetails.length
-        && this.allPresentStudentDetails.length === this.allCheckedStudentDetails.length;
-  }
-
-  get isAnyPresentCoursesChecked(): boolean {
-    return !!this.allPresentCourses.length && this.allCheckedCourses.length > 0;
-  }
-
-  courseStudentListSectionDataMap: { [key: string]: StudentListSectionData[] } = {};
-
-  constructor(private route: ActivatedRoute, private router: Router, private httpRequestService: HttpRequestService,
-              private statusMessageService: StatusMessageService) { }
-
   ngOnInit(): void {
     this.route.queryParams.subscribe((queryParams: any) => {
       this.user = queryParams.user;
-      this.loadCourses(this.user);
+      this.loadCourses();
     });
   }
 
   /**
-   * Navigate to the instructor search page with query input and query type: student.
-   */
-  search(): void {
-    this.router.navigate(['/web/instructor/search'], { queryParams: { studentSearchkey: this.searchQuery } });
-  }
-
-  /**
-   * Function to get initial data of the instructor's courses.
+   * Loads courses of current instructor.
    */
-  loadCourses(user: string): void {
-    const paramMap: { [key: string]: string } = { user };
-    this.httpRequestService.get('/instructor/students/courses', paramMap)
-        .subscribe((resp: GetCourseResponse) => {
-          this.courses = resp.courses;
-
-          if (!this.courses) {
-            this.statusMessageService.showWarningMessage('You do not have any courses yet.');
-          }
+  loadCourses(): void {
```

Comment:
In this method, you would populate `courseTabList`.

Reply:
Done.